programs/stake/src/stake_state.rs

```diff
@@ -842,6 +826,110 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+/// After calling `validate_delegated_amount()`, this struct contains calculated values that are used
+/// by the caller.
+struct ValidatedDelegatedInfo {
+    stake_amount: u64,
+}
+
+/// Ensure the stake delegation amount is valid.  This checks that the account meets the minimum
+/// balance requirements of delegated stake.  If not, return an error.
+fn validate_delegated_amount(
+    account: &KeyedAccount,
+    meta: &Meta,
+) -> Result<ValidatedDelegatedInfo, InstructionError> {
+    let stake_amount = account.lamports()?.saturating_sub(meta.rent_exempt_reserve); // can't stake the rent
```

Comment:
The function is called `validate_delegated_amount` but uses `saturating_sub` which won't throw an error if the balance is less than rent exempt reserve. Is it possible to use `checked_sub` instead of the saturating one and throw an error when the sub fails, or are there already stake accounts on chain which would fail in that case?

Reply:
The current behavior is intentional. (1) In the case where an account's balance is below the rent exempt minimum, then it has zero delegate-able lamports. (2) Adding new errors and/or changing what conditions return errors will change the program's behavior, which would break consensus without a feature-gate. 

(In an initial iteration of this PR I had checks within this function to error in these cases, but I learned that wouldn't be a good idea to add—for the above reasons.)

Reply:
OK, makes sense